src/poetry/core/packages/utils/link.py

```diff
@@ -136,6 +143,32 @@ def subdirectory_fragment(self) -> str | None:
 
     _hash_re = re.compile(r"(sha1|sha224|sha384|sha256|sha512|md5)=([a-f0-9]+)")
 
+    @property
+    def has_metadata(self) -> bool:
+        return self._metadata is not None and (self.is_wheel or self.is_sdist)
+
+    @property
+    def metadata_url(self) -> str | None:
+        if self.has_metadata:
+            return f"{self.url_without_fragment.split('?', 1)[0]}.metadata"
```

Comment:
Do you think a `url_without_query` property makes sense?

Reply:
Not necessarily for anything external. Maybe for internal use.